src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3227,6 +3227,132 @@ tx_gateway_frontend::get_all_transactions() {
       });
 }
 
+ss::future<result<tm_transaction, tx_errc>>
+tx_gateway_frontend::get_tx(kafka::transactional_id tid) {
+    auto shard = _shard_table.local().shard_for(model::tx_manager_ntp);
+
+    if (!shard.has_value()) {
+        vlog(txlog.warn, "can't find a shard for {}", model::tx_manager_ntp);
+        return ss::make_ready_future<result<tm_transaction, tx_errc>>(
+          tx_errc::shard_not_found);
+    }
+
+    return container().invoke_on(
+      *shard,
+      _ssg,
+      [tid](tx_gateway_frontend& self)
+        -> ss::future<result<tm_transaction, tx_errc>> {
+          auto partition = self._partition_manager.local().get(
+            model::tx_manager_ntp);
+          if (!partition) {
+              vlog(
+                txlog.warn,
+                "can't get partition by {} ntp",
+                model::tx_manager_ntp);
+              return ss::make_ready_future<result<tm_transaction, tx_errc>>(
+                tx_errc::partition_not_found);
+          }
+
+          auto stm = partition->tm_stm();
+
+          if (!stm) {
+              vlog(
+                txlog.error,
+                "can't get tm stm of the {}' partition",
+                model::tx_manager_ntp);
+              return ss::make_ready_future<result<tm_transaction, tx_errc>>(
+                tx_errc::stm_not_found);
+          }
+
+          auto gate_lock = gate_guard(self._gate);
```

Comment:
if the intention is to simply check if gate is closed please use `check()` instead. Otherwise the guard must be captured. Or you can use `ss::with_gate`

Reply:
It was with_gate
But then I rebase with problem